src/poetry/vcs/git/backend.py

```diff
@@ -176,7 +176,7 @@ def get_remote_url(repo: Repo, remote: str = "origin") -> str:
     @staticmethod
     def get_revision(repo: Repo) -> str:
         with repo:
-            return repo.head().decode("utf-8")
+            return repo.get_peeled(b"HEAD").decode("utf-8")
```

Comment:
@jelmer the return value of `head()` changed for tags. Previously the peeled value was returned, upgrading to 0.22.5 started breaking our integration tests.

Reply:
bound to be https://github.com/jelmer/dulwich/pull/1244, I bet

Reply:
That might be related to the symref regression that was fixed in 0.22.5.

Reply:
The issue was reproduced with 0.22.5 locally.

Reply:
To close the loop I have raised a dulwich issue for triage with a reproducer.

Cross-post: https://github.com/jelmer/dulwich/issues/1443